src/charm.py

```diff
@@ -1110,17 +1119,20 @@ def update_config(self) -> None:
             # then mark TLS as enabled. This commonly happens when the charm is deployed
             # in a bundle together with the TLS certificates operator. This flag is used to
             # know when to call the Patroni API using HTTP or HTTPS.
-            self.unit_peer_data.update({"tls": "enabled" if enable_tls else ""})
+            self.unit_peer_data.update({"tls": "enabled" if self.is_tls_enabled else ""})
             logger.debug("Early exit update_config: Patroni not started yet")
             return
 
-        restart_postgresql = enable_tls != self.postgresql.is_tls_enabled()
+        restart_postgresql = self.is_tls_enabled != self.postgresql.is_tls_enabled()
         self._patroni.reload_patroni_configuration()
-        self.unit_peer_data.update({"tls": "enabled" if enable_tls else ""})
+        self.unit_peer_data.update({"tls": "enabled" if self.is_tls_enabled else ""})
 
         # Restart PostgreSQL if TLS configuration has changed
         # (so the both old and new connections use the configuration).
         if restart_postgresql:
+            self.metrics_endpoint.update_scrape_job_spec(
+                self._generate_metrics_jobs(self.is_tls_enabled)
+            )
```

Comment:
If we are restarting due to TLS, we must also update the patroni endpoint scheme